src/core/ext/filters/client_channel/resolver/google_c2p/google_c2p_resolver.cc

```diff
@@ -173,7 +173,8 @@ void GoogleCloud2ProdResolver::StartLocked() {
           absl::StatusOr<std::string> result) mutable {
         resolver->work_serializer_->Run(
             [resolver, result = std::move(result)]() {
-              resolver->IPv6QueryDone(result.ok());
+              resolver->IPv6QueryDone(result.ok() ? std::move(result).value()
```

Comment:
Please add a comment explaining the reason for checking that the payload is non-empty, since it really is a fairly ugly hack that we're adding just to work around a bug outside of gRPC.

Reply:
done